doc/ref_model/chapters/chapter04.md

```diff
@@ -262,6 +263,32 @@ Table 4-10 shows performance measurement capabilities.
+
+**Table 4-12** shows Acceleration/Offload API Capabilities.
+
+<a name="Table4-12"></a>
+
+| Ref       | Acceleration/Offload API Capability | Unit   | Definition/Notes                                                                            |
+|-----------|--------------------------------------------|--------|---------------------------------------------------------------------------------------------|
+| e.api.001 | VNF/CNF usage of Accelerator standard i/f                           | Yes/No | VNF/CNF shall use abstracted standardized interfaces to the Acceleration/Offload function that enables multiple HW and SW implementation of the accelerated/offloaded function from multiple vendors in the Cloud Infrastructure                        |
+| e.api.002 | Virtualization Infrastructure SW usage of Accelerator standard i/f  | Yes/No | Virtualization Infrastructure SW shall use abstracted standardized interfaces to the HW-Acceleration/Offload  function that enables multiple HW and SW implementation in the HW Infrastructure Layer of the accelerated function from multiple vendors  |
+| e.api.003 | Accelerators offering standard i/f to HW Infra Layer                | Yes/No | Acceleration/Offload functions shall offer abstracted standardized interfaces for the Virtualization Infrastructure and VNF/CNF applications                                                                                                             |
+| e.api.004 | Accelerators offering virtualized functions                         | Yes/No | Acceleration/Offload functions for VNFs/CNFs should preferably be virtualized to allow multiple VNFs/CNFs to use the same Acceleration/Offload instance                                                                                                 |
+| e.api.005 | VNF/CNF Accelerator management functions access rights              | Yes/No | VNF/CNF management functions shall be capable to request Acceleration/Offload invocation without having elevated access rights                                                                                                                       |
+| e.api.006 | Accelerators offering standard i/f to VNF/CNF management            | Yes/No | VNF/CNF management functions should be capable to request Acceleration/Offload invocation through abstracted standardized Management interfaces                                                                                                       |
+| e.api.007 | VNFs/CNFs and Virtualization Infrastructure Accelerator portability | Yes/No | VNFs/CNFs and Virtualization Infrastructure SW should be designed to handle multiple types of Accelerator or Offload Function realizations even when their differences are exposed to the infrastructure or applications layers                    |
```

Comment:
Can you please clarify this requirement? Does this mean that each VNF/CNF should support a multiple types of accelerators, which are supported by the infrastructures?

Reply:
I can not know exactly which requirement text you are asking/commenting on, but I guess it is 007 and will answer as if that is the case:
Yes, an application and a Virtualization Infrastructure should be designed to make use of multiple different interfaces towards accelerators (and at least one SW implementation) when there are more common accelerators on the market. This is normally done through application certification and enable the applications to run on Cloud Infrastructure platforms that only have one or a selection of accelerators  available.
This is the norm in the IT and PC acceleration industry with GPU support being a good example.
The requirement do not say that  the situation of having vendor or chip specific interfaces is good, but recognize that this is often the case.

Reply:
The comment is related to e.api.007 (line 288).
Thanks for the answer, but does this leave the discovery of available accelerators and self-configuration to use one (or more) particular accelerator(s) to the application? What happens if the application gets migrated to a different host with different set of accelerators?

Reply:
This requirement do not define who would make a selection of what of the available accelerators that will actually be used, when there are more than one. In my view that should be an orchestrator (or even more specified a resource orchestrator), but it technically could also be the VNF/CNF. In my view the selection of where to use the available HW accelerators should be in the hands of an orchestrator that can optimize the actual deployment in the actual situation, honoring the needed characteristics.